lib/libzfs/libzfs_diff.c

```diff
@@ -478,7 +478,7 @@ differ(void *arg)
 	if (err)
 		return ((void *)-1);
 	if (di->zerr) {
-		ASSERT(di->zerr == EINVAL);
+		ASSERT(di->zerr == EPIPE);
```

Comment:
Thanks, the updated code looks right and I'm OK with it as is.  But I wonder if it wouldn't be better to instead remove the ASSERT and update the error message to log the `di->err`.  This way it's still useful even in non-debug builds.

Reply:
The value of `di->zerr` will only ever be EPIPE in this context, and the assertion serves to document that fact. The purpose of this special case is to give a more relevant internal error message in the context of `differ()`. I prefer the ASSERT since it informs the developer that any other value here would be a programming error, not a conceivable run-time error.